### PR TITLE
Fix final board postcondition

### PR DESCRIPTION
**Pull request by rodri042**
Fixes https://github.com/mumuki/mumuki-gobstones-runner/issues/15
Fixes https://github.com/mumuki/mumuki-gobstones-runner/issues/16

### CODE REVIEW

File: lib/gobstones/batch_parser.rb [hunk 1]
```diff
@@ -43,9 +43,17 @@ def parse_options(test)
       [
         struct(key: :show_initial_board, default: true),
         struct(key: :show_final_board, default: true),
-        struct(key: :check_head_position, default: false),
+        struct(key: :check_head_position, default: true),
```
Comment (flbulgarelli) [inline lib/gobstones/batch_parser.rb]: I am not sure of this change. Which is the default in xgobstones-runner?

Although It looks fine to me, I would like to avoid exercise incompatibilities.

File: lib/gobstones/batch_parser.rb [hunk 2]
```diff
@@ -43,9 +43,17 @@ def parse_options(test)
...
         struct(key: :subject, default: nil)
-      ].map { |it| [it.key, test[it.key] || it.default] }.to_h
+      ].map { |it| [
+          it.key,
+          if test[it.key].nil?
```
Comment (flbulgarelli) [inline lib/gobstones/batch_parser.rb]: Why this change? Can the key be `false`?

File: lib/gobstones/batch_parser.rb (hunk 1 above)
Comment (rodri042) [reply lib/gobstones/batch_parser.rb]: The default was `false`. I just changed it again to that value.

File: lib/gobstones/batch_parser.rb (hunk 2 above)
Comment (rodri042) [reply lib/gobstones/batch_parser.rb]: This wasn't working well. For example, the initial value of `show_final_board` is true. If the YML had `false` in the YML, it was running `false || true` resulting in `true` again.

Comment (flbulgarelli) [reply lib/gobstones/batch_parser.rb]: OK, I see. It is a pity that there is a default whose value is `true` .

Comment (flbulgarelli) [reply lib/gobstones/batch_parser.rb]: In the future, i also recommend using a ternary if `...?...:...`
